File: backend/app/schemas/progress.py
from pydantic import BaseModel


class SubjectProgress(BaseModel):
    subject: str
    quizzes_taken: int
    average_score: int
    topics_explained: int


class ActivityItem(BaseModel):
    type: str
    subject: str
    topic: str
    detail: str
    created_at: str


class ProgressSummary(BaseModel):
    total_quizzes: int
    average_score: int
    total_topics_explained: int
    by_subject: list[SubjectProgress]
    recent_activity: list[ActivityItem]

File: frontend/src/types/progress.ts
export interface SubjectProgress {
  subject: string
  quizzesTaken: number
  averageScore: number
  topicsExplained: number
}

export interface ActivityItem {
  type: 'quiz' | 'topic'
  subject: string
  topic: string
  detail: string
  createdAt: string
}

export interface ProgressSummary {
  totalQuizzes: number
  averageScore: number
  totalTopicsExplained: number
  bySubject: SubjectProgress[]
  recentActivity: ActivityItem[]
}
